src/order/directed.lean

```diff
@@ -271,3 +250,34 @@ instance order_top.to_is_directed_le [has_le α] [order_top α] : is_directed α
 @[priority 100]  -- see Note [lower instance priority]
 instance order_bot.to_is_directed_ge [has_le α] [order_bot α] : is_directed α (≥) :=
 ⟨λ a b, ⟨⊥, bot_le, bot_le⟩⟩
+
+section scott_continuous
+
+variables [preorder α] {a : α}
+
+/--
+A function between preorders is said to be Scott continuous if it preserves `is_lub` on directed
+sets. It can be shown that a function is Scott continuous if and only if it is continuous wrt the
+Scott topology.
+-/
+def scott_continuous [preorder β] (f : α → β) : Prop :=
+∀ (d : set α) (a : α), d.nonempty → directed_on (≤) d → is_lub d a → is_lub (f '' d) (f a)
```

Comment:
This name seems to conflict with the docstring of `omega_complete_partial_order.continuous`. Can the two notions be unified or are they genuinely different?

Reply:
@YaelDillies from https://github.com/leanprover-community/mathlib4/pull/2508

> There is already a related notion of Scott topology defined in topology.omega_complete_partial_order, where it is defined on ω-complete partial orders in terms of ω-chains. In some circumstances the definition given here coincides with that given in topology.omega_complete_partial_order but in general they are different. Abramsky and Jung ([Domain Theory, 2.2.4][abramsky_gabbay_maibaum_1994]) argue that the ω-chain approach has pedagogical advantages, but the directed sets approach is more appropriate as a theoretical foundation.

Reply:
See: https://www.cs.bham.ac.uk/~axj/pub/papers/handy1.pdf

Reply:
Ahah, I see.